src/main/groovy/com/calgaryscientific/gradle/VeracodeWorkflowSandboxTask.groovy

```diff
@@ -0,0 +1,125 @@
+/*******************************************************************************
+ * MIT License
+ *
+ * Copyright (c) 2017 Calgary Scientific Incorporated
+ *
+ * Copyright (c) 2013-2014 kctang
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ ******************************************************************************/
+
+package com.calgaryscientific.gradle
+
+import groovy.transform.CompileStatic
+
+@CompileStatic
+class VeracodeWorkflowSandboxTask extends VeracodeTask {
+    static final String NAME = 'veracodeSandboxWorkflow'
+    String build_version
+    String maxUploadAttempts
+    String waitTimeBetweenAttempts
+    String delete
+
+    VeracodeWorkflowSandboxTask() {
+        group = 'Veracode Sandbox'
+        description = "Run through the Veracode Workflow for the given 'app_id' and 'sandbox_id' using 'build_version' as the build identifier"
+        requiredArguments << 'app_id' << 'sandbox_id' << 'build_version'
+        optionalArguments << 'maxUploadAttempts' << 'waitTimeBetweenAttempts' << 'delete'
+        app_id = project.findProperty("app_id")
+        sandbox_id = project.findProperty("sandbox_id")
+        build_version = project.findProperty("build_version")
+    }
+
+    VeracodeGetBuildInfoSandboxTask buildInfoSandboxTask = new VeracodeGetBuildInfoSandboxTask()
```

Comment:
this gradle tasks depends on a whole bunch of other gradle tasks is a little bit worrisome to me....

Reply:
In what way is dependency worrisome? Can you be more explicit please?

This task is a task that runs the entire workflow so it basically orchestrates running the other tasks. I would not want to duplicate that behaviour. But I don't really know what your concern is.

Reply:
- sorry, I didn't clear with my concerns, as far as I understand, gradle custom task were not designed to compose other tasks, in case you need to do it, you can use aggregate tasks with dependsOn....
- what I'm getting at is you're probably aware that I'm not a big fan of putting too much logic/stuff into gradle custom task class as this is the API between gradle and our custom business logic. It would make it harder to test custom business logic
  - if we include business logic in the gradle custom task as it relies on gradle support such as testkit or gradle tools to support testing gradle custom tasks. 
  - What I have seen over the years is recommendation to separate the domain logic from technology details as much as possible. The main reason for that is business logic has to live longer than the technology, what if gradle deprecated, what if we want to use something else other than gradle.
  - We should build our product in such a way that it's easy to switch to different technologies.....